### PR TITLE
benchadapt: log to stderr instead of stdout

### PR DESCRIPTION
**Pull request by jgehrcke**
For issue #805.

### CODE REVIEW

File: benchclients/python/benchclients/logging.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import logging
 
+log = logging.getLogger(__name__)
+
 logging.basicConfig(
-    format="%(levelname)s [%(asctime)s] %(message)s", level=logging.INFO
```
Comment (jgehrcke) [inline benchclients/python/benchclients/logging.py:4]: compared to the 'old' format this adds fractions of a second with millisecond resolution: `%(msecs)03d`

File: benchadapt/python/benchadapt/adapters/_adapter.py [hunk 2]
```diff
@@ -1,15 +1,28 @@
...
+
+# `basicConfig()` does nothing if the root logger already has handlers
+# configured (that is, if this benchadapt library is imported in the context
+# of a program that already has a root logger with handlers set up, then the
+# following call is a noop.
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s.%(msecs)03d %(levelname)s: %(message)s",
+    datefmt="%y%m%d-%H:%M:%S",
+)
```
Comment (jgehrcke) [inline benchadapt/python/benchadapt/adapters/_adapter.py:24]: This takes effect if the benchadapt library used from within any program that is not `benchclients`.

File: benchclients/python/benchclients/logging.py [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
+    level=logging.INFO,
+    format="%(asctime)s.%(msecs)03d %(levelname)s: %(message)s",
+    datefmt="%y%m%d-%H:%M:%S",
```
Comment (jgehrcke) [inline benchclients/python/benchclients/logging.py:8]: This takes effect if the benchadapt library is used from within `benchclients`.

Comment (jgehrcke) [reply benchclients/python/benchclients/logging.py:8]: Is benchclients a CLI? Or another set of libraries?

Comment (alistaire47) [reply benchclients/python/benchclients/logging.py:8]: Libraries. The only one that's a CLI is benchconnect

File: benchadapt/python/benchadapt/adapters/_adapter.py [hunk 4]
```diff
@@ -190,7 +203,7 @@ def post_results(self) -> list:
             res_list.append(res)
 
         if error:
-            print(f"POST at {error_time} timed out twice:")
+            log.error(f"POST at {error_time} timed out twice:")
             raise error
```
Comment (alistaire47) [inline benchadapt/python/benchadapt/adapters/_adapter.py:207]: If there's value in this retry-on-timeout code, it should get migrated to benchclients so all requests get more robust handling, not just these particular [important!] ones. Made #824 for this

File: benchclients/python/benchclients/logging.py (hunk 3 above)
Comment (alistaire47) [reply benchclients/python/benchclients/logging.py:8]: ...and datalogistik, but I think that's still in a separate repo for now

**Review by alistaire47 — APPROVED** [review]
Looks good. I think in a lot of contexts users will want the benchclients log set at DEBUG so all request bodies get logged, but it's fine to let users opt-in to that.

File: benchclients/python/benchclients/logging.py (hunk 3 above)
Comment (jgehrcke) [reply benchclients/python/benchclients/logging.py:8]: Thanks!

**Comment by jgehrcke** [comment]
>  I think in a lot of contexts users will want the benchclients log set at DEBUG so all request bodies get logged,

not sure, are you implying that you think that behavior changed in this regard via this PR?

**Pull request merged**